app/assets/javascripts/segment_history_graph.coffee

```diff
@@ -15,7 +15,7 @@ $ ->
   hidden_data = []
   first_non_null_attempt = Infinity
 
-  $.each gon.run.splits, (i, split) ->
+  $.each gon.run.raw_splits, (i, split) ->
```

Comment:
I'm not exactly sure why it changes what populates gon.run.splits to be quite honest.  Checking the contents of it pre-commits shows that the history is present in gon.run.splits, but when the logic is moved out of the controller it no longer puts the history in gon.run.splits.  I'm not sure if there is a call to some function somewhere that I'm just missing.